config/calibrate.py

```diff
@@ -2,26 +2,16 @@
 import os.path
 
 from lsst.utils import getPackageDir
-from lsst.meas.algorithms import LoadIndexedReferenceObjectsTask
 
 hscDir = os.path.join(getPackageDir("obs_subaru"), "config")
 
 # Use gaia for astrometry (phot_g_mean for everything, as that is the broadest
-# band with the most depth)
-# Use panstarrs for photometry (grizy filters)
-for refObjLoader in (config.astromRefObjLoader,
-                     config.photoRefObjLoader,):
-    refObjLoader.retarget(LoadIndexedReferenceObjectsTask)
-    # HSC has *lots* of filters, easier to modify the official config
-    refObjLoader.load(os.path.join(hscDir, "filterMap.py"))
-    # Official config doesn't bother mapping ugrizy names, which are used in Gen 2
-    for hscFilter in ["u", "g", "r", "i", "z", "y"]:
-        config.astromRefObjLoader.filterMap[hscFilter] = hscFilter
```

Comment:
Can you explain why you removed this? All the dataset configs explicitly set the `refObjLoader` tasks so that we're configuring the task we expect to be, even if the default or the obs config changes out from under us. The fact that at least two completely different changes are being blended (and "and"ed) in the same commit doesn't help.

Reply:
`LoadIndexedReferenceObjectsTask` is a gen2-only thing, and I had to unravel the `for` loop because the astrometry and photometry refcats use different configs (no filterMap for Gaia and none needed for ps1).

Reply:
Ah, [so I misread the Gen 3 code](https://github.com/lsst/pipe_tasks/blob/main/python/lsst/pipe/tasks/calibrate.py#L636-L640). Would it be useful to explicitly retarget to `ReferenceObjectLoader`, for the same reasons as above?

In any case, if you're changing this part of the config, you should also change it for the HiTS datasets.

Reply:
ReferenceObjectLoader isn't a Task, it happens during QG generation, so there's nothing to retarget.

Good point on changing the HITS datasets, too.